dvc/fs/webhdfs.py

```diff
@@ -123,7 +123,7 @@ def exists(self, path_info) -> bool:
 
     def info(self, path_info):
         st = self.hdfs_client.status(path_info.path)
-        return {"size": st["length"]}
+        return {"size": st["length"], "type": "file"}
```

Comment:
I guess we could try to use `type` from the status call here?

Reply:
For providers that lack of support for directories (gdrive, webdhfs) it is a static. Otherwise it would change the current behavior for staging (in the past `isdir()` would always return `False` for the same path, now `info()['type'] == 'directory'` returning True).

Reply:
webhdfs doesn't have directories? O_o Am I missing something?

Reply:
There is no `isdir()` method implemented, even though there are dirs in webhdfs we don't support them.

Reply:
@isidentical Ah, gotcha. So you are just matching the issues with our old implementation? :slightly_smiling_face: If we can report proper type here with reasonable effort - we should, or we run a chance of running into this odd bug in the future. I guess we need to finally start migrating to fsspec's webhdfs/hdfs implementations...

Reply:
Yes. This is a best-effort optimization PR, so I did not try to implement anything new or change any behavior.